### PR TITLE
Add get random seed method

### PR DESCRIPTION
**Pull request by hunkim98**
Add get random seed method

Resolves issue #2131

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -55,9 +55,22 @@ To be released.
...
+=======
+[#1972]: https://github.com/planetarium/libplanet/issues/1972
+[#2179]: https://github.com/planetarium/libplanet/pull/2179
+>>>>>>> f53411653 (Add genesis hash property to IActionContext)
+=======
+[#1972]: https://github.com/planetarium/libplanet/issues/1972
+=======
+>>>>>>> 331030a8d (Apply reviews)
+[#2179]: https://github.com/planetarium/libplanet/pull/2179
+>>>>>>> f0d0f666a (Add test code and add info to changes.md)
```
Comment (longfin) [inline CHANGES.md:73]:
```suggestion
[#1972]: https://github.com/planetarium/libplanet/issues/1972
[#2173]: https://github.com/planetarium/libplanet/pull/2173
[#2179]: https://github.com/planetarium/libplanet/pull/2179
[#2183]: https://github.com/planetarium/libplanet/pull/2183
```

File: Libplanet.Tests/Action/ActionEvaluatorTest.cs [hunk 2]
```diff
@@ -1100,6 +1100,85 @@ private async Task CheckGenesisHashInAction()
...
+                0x97, 0x44, 0x1f, 0xbf, 0xa2, 0x5c, 0x8b, 0xd0, 0x0e, 0x23, 0x02, 0x20, 0x3c,
+                0x06, 0x02, 0x1f, 0xb8, 0x3f, 0x67, 0x49, 0x92, 0x3c, 0x07, 0x59, 0x67, 0x96,
+                0xa8, 0x63, 0x04, 0xb0, 0xc3, 0xfe, 0xbb, 0x6c, 0x7a, 0x7b, 0x58, 0x58, 0xe9,
+                0x7d, 0x37, 0x67, 0xe1, 0xe9,
+            };
+            byte[] hashedSignature;
+            using (var hasher = SHA1.Create())
+            {
+                hashedSignature = hasher.ComputeHash(signature);
+            }
```
Comment (longfin) [inline Libplanet.Tests/Action/ActionEvaluatorTest.cs:1124]:
```suggestion
            using SHA1 hasher = SHA1.Create())
            hashedSignature = hasher.ComputeHash(signature);
```

[using statement](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement) seems enough in this case.

File: Libplanet.Tests/Action/ActionEvaluatorTest.cs [hunk 3]
```diff
@@ -1100,6 +1100,85 @@ private async Task CheckGenesisHashInAction()
...
 
+        [Fact]
+        private void GenerateRandomSeed()
+        {
+            byte[] preEvaluationHashBytes =
+            {
+                0x45, 0xa2, 0x21, 0x87, 0xe2, 0xd8, 0x85, 0x0b, 0xb3, 0x57,
+                0x88, 0x69, 0x58, 0xbc, 0x3e, 0x85, 0x60, 0x92, 0x9c, 0xcc,
+            };
+            var signature = new byte[]
```
Comment (longfin) [inline Libplanet.Tests/Action/ActionEvaluatorTest.cs:1111]:
```suggestion
            byte[] signature =
```
Just for consistency with above😅

**Review by longfin — CHANGES_REQUESTED** [review]
And, we need to a mention about `ActionEvaluator<T>.GenerateRandomSeed()` in `CHANGES.md`. 😄

**Comment by hunkim98** [comment]
Changes.md has been modified. GetRandomSeed static method has been added under Added APIs

File: CHANGES.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ To be released.
 ### Backward-incompatible storage format changes
 
 ### Added APIs
-
+ -  Added `ActionEvaluator<T>.GenerateRandomSeed()` static method.  [[#2236]]
```
Comment (colibrishin) [inline CHANGES.md:18]:
```suggestion
 -  Added `ActionEvaluator<T>.GenerateRandomSeed()` static method.  [[#2236]]
 
```

File: CHANGES.md [hunk 5]
```diff
@@ -388,6 +390,7 @@ Released on August 12, 2022.
 [#2159]: https://github.com/planetarium/libplanet/pull/2159
 [#2162]: https://github.com/planetarium/libplanet/issues/2162
 [#2171]: https://github.com/planetarium/libplanet/pull/2171
+[#1972]: https://github.com/planetarium/libplanet/issues/1972
```
Comment (longfin) [inline CHANGES.md:393]:
```suggestion
```

File: CHANGES.md [hunk 6]
```diff
@@ -35,6 +36,7 @@ To be released.
 
 ### CLI tools
 
+[#2236]: https://github.com/planetarium/libplanet/pull/2236
```
Comment (longfin) [inline CHANGES.md:39]:
```suggestion
[#2131]: https://github.com/planetarium/libplanet/issues/2131
[#2236]: https://github.com/planetarium/libplanet/pull/2236
```

File: CHANGES.md [hunk 7]
```diff
@@ -23,6 +23,7 @@ To be released.
  -  Added `BlockChain<T>.MakeTransaction(PrivateKey, IAction,
     IImmutableSet<Address>, DateTimeOffset?)` overloaded method.
     [[#2151], [#2273]]
+ -  Added `ActionEvaluator<T>.GenerateRandomSeed()` static method.  [[#2236]]
```
Comment (longfin) [inline CHANGES.md:26]:
```suggestion
 -  Added `ActionEvaluator<T>.GenerateRandomSeed()` static method.  [[#2131], [#2236]]
```

**Review by longfin — DISMISSED** [review]

File: Libplanet.Tests/Action/ActionEvaluatorTest.cs [hunk 8]
```diff
@@ -1104,6 +1104,83 @@ private async Task CheckGenesisHashInAction()
...
+            {
+                0x30, 0x44, 0x02, 0x20, 0x2f, 0x2d, 0xbe, 0x5a, 0x91, 0x65, 0x59, 0xde, 0xdb,
+                0xe8, 0xd8, 0x4f, 0xa9, 0x20, 0xe2, 0x01, 0x29, 0x4d, 0x4f, 0x40, 0xea, 0x1e,
+                0x97, 0x44, 0x1f, 0xbf, 0xa2, 0x5c, 0x8b, 0xd0, 0x0e, 0x23, 0x02, 0x20, 0x3c,
+                0x06, 0x02, 0x1f, 0xb8, 0x3f, 0x67, 0x49, 0x92, 0x3c, 0x07, 0x59, 0x67, 0x96,
+                0xa8, 0x63, 0x04, 0xb0, 0xc3, 0xfe, 0xbb, 0x6c, 0x7a, 0x7b, 0x58, 0x58, 0xe9,
+                0x7d, 0x37, 0x67, 0xe1, 0xe9,
+            };
+            byte[] hashedSignature;
+            using SHA1 hasher = SHA1.Create())
```
Comment (riemannulus) [inline Libplanet.Tests/Action/ActionEvaluatorTest.cs:1125]: This is a typo, right?
```suggestion
            using SHA1 hasher = SHA1.Create()
```

**Review by riemannulus — CHANGES_REQUESTED** [review]

**Review by longfin — APPROVED** [review]

File: CHANGES.md [hunk 9]
```diff
@@ -23,6 +23,7 @@ To be released.
  -  Added `IRoutingTable` interface.  [[#2046], [#2229]]
  -  `RoutingTable` became to implement `IRoutingTable` interface.
     [[#2046], [#2229]]
+ -  Added `ActionEvaluator<T>.GenerateRandomSeed()` static method.  [[#2131], [#2236]]
```
Comment (longfin) [inline CHANGES.md:26]:
```suggestion
 -  Added `ActionEvaluator<T>.GenerateRandomSeed()` static method.  
    [[#2131], [#2236]]
```

**Review by riemannulus — APPROVED** [review]

**Comment by riemannulus** [comment]
PTAL 🙏 @longfin @colibrishin

**Review by colibrishin — APPROVED** [review]
LGTM

**Review by longfin — APPROVED** [review]

**Pull request merged**